From 1cea3daee72ea75c6bf8199485e955a43526e246 Mon Sep 17 00:00:00 2001
From: Ping Sun <ping.x.sun@sony.com>
Date: Mon, 7 Jan 2019 10:01:49 +0800
Subject: [PATCH 4/7] Fix the drawable resource can't be loaded from RRO
 package

Since refer to resources from the overlay package using the IDs defined
in the original package, NotFoundException occurs and icons can NOT be
loaded.
These are caused by using Icon.createWithResource(Resources, ResId),
and usage of this API is NOT supported.

To resolve this issue,  use Icon.createWithResource(Context, ResId) for
create icons with original package.

Test: manual - Check that SIP call account icon is displayed with the
specified in the overlay package
Bug: 124352738

Change-Id: I43be48818b190698a4344d8fc8f2ac5a6927702a
---
 sip/src/com/android/services/telephony/sip/SipUtil.java     | 2 +-
 src/com/android/services/telephony/ImsConference.java       | 3 +--
 src/com/android/services/telephony/TelephonyConnection.java | 3 +--
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/sip/src/com/android/services/telephony/sip/SipUtil.java b/sip/src/com/android/services/telephony/sip/SipUtil.java
index d67422568..ff38754db 100644
--- a/sip/src/com/android/services/telephony/sip/SipUtil.java
+++ b/sip/src/com/android/services/telephony/sip/SipUtil.java
@@ -134,7 +134,7 @@ public class SipUtil {
                 .setAddress(sipUri)
                 .setShortDescription(sipAddress)
                 .setIcon(Icon.createWithResource(
-                        context.getResources(), R.drawable.ic_dialer_sip_black_24dp))
+                        context, R.drawable.ic_dialer_sip_black_24dp))
                 .setExtras(phoneAccountExtras)
                 .setSupportedUriSchemes(supportedUriSchemes);
 
diff --git a/src/com/android/services/telephony/ImsConference.java b/src/com/android/services/telephony/ImsConference.java
index c62a7f10d..fcf1fe651 100644
--- a/src/com/android/services/telephony/ImsConference.java
+++ b/src/com/android/services/telephony/ImsConference.java
@@ -990,8 +990,7 @@ public class ImsConference extends Conference implements Holdable {
                 setStatusHints(new StatusHints(
                         context.getString(R.string.status_hint_label_wifi_call),
                         Icon.createWithResource(
-                                context.getResources(),
-                                R.drawable.ic_signal_wifi_4_bar_24dp),
+                                context, R.drawable.ic_signal_wifi_4_bar_24dp),
                         null /* extras */));
             }
         } else {
diff --git a/src/com/android/services/telephony/TelephonyConnection.java b/src/com/android/services/telephony/TelephonyConnection.java
index cd55a3bf6..2c5004941 100644
--- a/src/com/android/services/telephony/TelephonyConnection.java
+++ b/src/com/android/services/telephony/TelephonyConnection.java
@@ -2050,8 +2050,7 @@ abstract class TelephonyConnection extends Connection implements Holdable {
             setStatusHints(new StatusHints(
                     context.getString(labelId),
                     Icon.createWithResource(
-                            context.getResources(),
-                            R.drawable.ic_signal_wifi_4_bar_24dp),
+                            context, R.drawable.ic_signal_wifi_4_bar_24dp),
                     null /* extras */));
         } else {
             setStatusHints(null);
-- 
2.17.1

